da/mock/mock.go

```diff
@@ -68,6 +87,51 @@ func (m *DataAvailabilityLayerClient) Stop() error {
 	return nil
 }
 
+// GetHeaderByHeight returns the header at the given height.
+func (m *DataAvailabilityLayerClient) GetHeaderByHeight(height uint64) *core.DataAvailabilityHeader {
+	m.daHeadersLock.RLock()
+	dah := m.daHeaders[height]
```

Comment:
Why are are not checking for existence in the map?
This should be explained in the method docstring or updated to handle the case where it doesn't exist.